core/annotations/annotations.go

```diff
@@ -11,15 +11,15 @@ import (
 )
 
 // Kind identifies different kinds of entities that'll get associated with annotations.
-type Kind int
+type Kind string
 
 const (
-	KindCharm Kind = iota
-	KindMachine
-	KindUnit
-	KindModel
-	KindVolume
-	KindFilesystem
+	KindApplication Kind = "application"
+	KindCharm       Kind = "charm"
+	KindMachine     Kind = "machine"
+	KindUnit        Kind = "unit"
+	KindModel       Kind = "model"
+	KindStorage     Kind = "storage_instance"
```

Comment:
I don't think we should tie these to the database tables. There should be a function in the state layer that converts a Kind to a table.